### PR TITLE
fix: record select source locations are wrong

### PR DESCRIPTION
**Pull request by LoZander**
This PR fixes an issue where the `SourceLocation` of `RecordSelect`s are wrong. The issue is that the beginning of the `SourceLocation`s currently start at the selected field instead of the record expression.

For instance, for
```flix
r#x
```
the `SourceLocation` only includes `x`.

This is particularly an issue for #9114.

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala [hunk 1]
```diff
@@ -1608,7 +1608,7 @@ object Weeder2 {
       mapN(pickExpr(tree)) {
         expr =>
           idents.foldLeft(expr) {
-            case (acc, ident) => Expr.RecordSelect(acc, Name.mkLabel(ident), ident.loc)
+            case (acc, ident) => Expr.RecordSelect(acc, Name.mkLabel(ident), tree.loc)
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala:1611]: does this fix work for `r#x#y#z`?

Comment (LoZander) [reply main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala:1611]: Yeah, you're right. I think I've fixed it. We want the location that from tree.loc through ident.loc for each

Comment (LoZander) [reply main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala]: I believe I've fixed it now. I've tested it on a nested example (similar to what you proposed. Good catch. Before they all had the same sourcelocation, which ofc. makes no sense.

**Comment by LoZander** [comment]
Not sure why the IDE tests are failing. But it doesn't seem related to this

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**